src/flexmeasures_client/s2/__init__.py

```diff
@@ -0,0 +1,211 @@
+        @register
+        """
+        for method in dir(self):
+            if not method.startswith("__") and hasattr(getattr(self, method), "_tag"):
+                tag = getattr(getattr(self, method), "_tag")
+                self.message_handlers[tag.message_type] = getattr(self, method)
+
+    def register_callback(
+        self, callback_store: dict, message_id: str, callback: Callable, **kwargs
+    ):
+        """
+        Stores a callback into a callback store indexed by the message.
+        """
+        callback_store[message_id] = functools.partial(callback, **kwargs)
+
+    def register_success_callback(self, message_id: str, callback: Callable, **kwargs):
+        """
+        Stores a callback into the success callback store.
+        Callbacks will be called if ReceptionStatus.satus = ReceptionStatusValues.OK
+        """
+        self.register_callback(self.success_callback, message_id, callback, **kwargs)
+
+    def register_failiure_callback(self, message_id: str, callback: Callable, **kwargs):
+        """
+        Stores a callback into the failiure callback store.
+        Callbacks will be called if ReceptionStatus.satus != ReceptionStatusValues.OK
+        """
+        self.register_callback(self.failiure_callback, message_id, callback, **kwargs)
+
+    def supports_message(self, message: Dict | pydantic.BaseModel) -> bool:
+        """
+        Checks if a message is supported by the Handler.
+        """
+
+        # TODO: handle a message that has not `message_type`
+        message_type = ""
+
+        if isinstance(message, pydantic.BaseModel):
+            message_type = message.message_type
+        else:
+            message_type = message.get("message_type")
+
+        return message_type in self.message_handlers
+
+    def handle_message(self, message: pydantic.BaseModel) -> dict:
+        """
+        Calls the handler linked to the message_type and converts the output
+        to a serilizd dict, i.e, it converts all the inner objects to Python
+        basic types (dict, list, int, str, ...).
+
+        :returns serialized_output_message:
+        """
+
+        output_message = self.message_handlers[message.get("message_type")](message)
+
+        # Pending for pydantic V2 to implemen model.model_dump(mode="json") in
+        # PR #1409 (https://github.com/pydantic/pydantic/issues/1409)
+        if output_message:  # e.g of return None is handling ReceptionStatus
+            output_message = json.loads(output_message.json())
+
+        return output_message
+
+    @register(ReceptionStatus, "ReceptionStatus")
+    def handle_response_status(self, message: ReceptionStatus):
+        """
+        If defined, it calls the success_callback or failiure_callback depeding
+        on the status of the ReceptionStatus.
+
+        By default, the handlers will use this handler for the messages of
+        type ReceptionStatus.
+        """
+
+        callback_store = None
+
+        # saving ACK status code
```

Comment:
Also, as a general comment, also relating to your docstrings: [PEP 257](https://peps.python.org/pep-0257/) recommends prescribing effects as commands: `save status` rather than `saves status`, or `saving status`.